git commit -m "add Diplom_1"

MariF24 opened the pull request.

MariF24 committed:
git commit -m "add Diplom_1"

AlexKitIt commented on tests/test_buns.py:
```diff
@@ -0,0 +1,15 @@
+from data import bun_price_correct, bun_name_correct, bun_name_incorrect, bun_price_incorrect
+class TestBuns:
+    def test_get_correct_bun(self):
+        bun = Bun(bun_name_correct, bun_price_correct)
+
+        assert bun.name == bun_name_correct
+        assert bun.price == bun_price_correct
+
+    def test_get_incorrect_bun(self):
+        bun = Bun(bun_name_incorrect, bun_price_incorrect)
+
+        assert bun.get_name != bun_name_incorrect
+        assert bun.get_price != bun_price_incorrect
```
Нужно исправить: тут должны быть проверки get_name и get_price

AlexKitIt commented on tests/test_burger.py:
```diff
@@ -0,0 +1,80 @@
+        burger.add_ingredient(ingredient_first)
+        burger.add_ingredient(ingredient_double)
+
+        assert '==== Bulka ====' in burger.get_receipt()
```
Нужно исправить: вывод метода слишком сложен, чтобы просто проверять его по in, стоит проверять по хардкоженной строке полного рецепта

AlexKitIt commented on tests/test_burger.py:
```diff
@@ -0,0 +1,80 @@
+
+
+    def test_get_price_true(self):
+        mock_bun = Mock()
+        mock_bun.get_name.return_value = bun_name_correct
+        mock_bun.get_price.return_value = bun_price_correct
```
Эта чать кода повторяется из теста в тест, значит что это предусловие и его стоит описать в фикстуре

AlexKitIt commented on tests/test_ingredient.py:
```diff
@@ -0,0 +1,29 @@
+class TestIngredients:
+
+
+    def test_get_correct_ingredient(self):
+        ingredient = Ingredient(ingredient_type_first_correct, ingredient_name_first_correct, ingredient_price_first_correct)
+
+        assert ingredient.type == ingredient_type_first_correct
+        assert ingredient.name == ingredient_name_first_correct
+        assert ingredient.price == ingredient_price_first_correct
+
+
+    def test_get_incorrect_ingredient(self):
+        ingredient = Ingredient(ingredient_type_incorrect, ingredient_name_incorrect, ingredient_price_incorrect)
+
+        assert ingredient.get_type != ingredient_type_incorrect
+        assert ingredient.get_name != ingredient_name_incorrect
```
Нужно исправить: тут должны быть проверки методов get_type get_name get_price

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"
git commit -m "add index.html"

AlexKitIt commented on tests/test_buns.py:
```diff
@@ -0,0 +1,10 @@
+from praktikum.bun import Bun
+from data import bun_price, bun_name
+class TestBuns:
+    def test_get_bun(self):
+        bun = Bun(bun_name, bun_price)
+        bun.get_name()
+        bun.get_price()
+        assert bun.name == bun_name
+        assert bun.price == bun_price
```
Нужно исправить: это юнит тесты, один тест - один проверяемый метод. Тут тестов должно быть два

AlexKitIt commented on tests/test_ingredient.py:
```diff
@@ -0,0 +1,23 @@
+        assert ingredient.type == ingredient_type_first
+        assert ingredient.name == ingredient_name_first
+        assert ingredient.price == ingredient_price_first
+
```
Нужно исправить:а тут три теста

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"
git commit -m "add index.html"
git commit -m "add Diplom_1_v.3"

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"
git commit -m "add index.html"
git commit -m "add Diplom_1_v.3"
git commit -m "add Diplom_1_v.3_2"

AlexKitIt commented on tests/test_buns.py:
```diff
@@ -0,0 +1,16 @@
+class TestBuns:
+    def test_get_bun_name(self):
+        bun = Bun(bun_name, bun_price)
+        bun.get_name()
+
+        assert bun.name == bun_name
+
+
+    def test_get_bun_price(self):
+        bun = Bun(bun_name, bun_price)
+        bun.get_price()
+
+        assert bun.price == bun_price
```
Нужно исправить: эти методы возвращают значения, но они никуда не записываются, а по тесту должны проверяться возвращаемые ими значения

AlexKitIt commented on tests/test_ingredient.py:
```diff
@@ -0,0 +1,29 @@
+
+    def test_get_ingredient_type(self):
+        ingredient = Ingredient(ingredient_type_first, ingredient_name_first, ingredient_price_first)
+        ingredient.get_type()
+
+        assert ingredient.type == ingredient_type_first
+
+
+    def test_get_ingredient_name(self):
+        ingredient = Ingredient(ingredient_type_first, ingredient_name_first, ingredient_price_first)
+        ingredient.get_name()
+
+        assert ingredient.name == ingredient_name_first
+
+
+    def test_get_ingredient_price(self):
+        ingredient = Ingredient(ingredient_type_first, ingredient_name_first, ingredient_price_first)
+        ingredient.get_price()
+
+        assert ingredient.price == ingredient_price_first
```
Нужно исправить: аналогично верхнему комменту. Зачем вызываются методы если не проверяется их работа?

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"
git commit -m "add index.html"
git commit -m "add Diplom_1_v.3"
git commit -m "add Diplom_1_v.3_2"
git commit -m "add Diplom_1_v.4"

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"
git commit -m "add index.html"
git commit -m "add Diplom_1_v.3"
git commit -m "add Diplom_1_v.3_2"
git commit -m "add Diplom_1_v.4"
git commit -m "add Diplom_1_v.5"

MariF24 committed:
git commit -m "add Diplom_1"
git commit -m "add Diplom_1_v.2"
git commit -m "add index.html"
git commit -m "add Diplom_1_v.3"
git commit -m "add Diplom_1_v.3_2"
git commit -m "add Diplom_1_v.4"
git commit -m "add Diplom_1_v.5"
git commit -m "add Diplom_1_v.6"
